packages/core/src/actions/contracts/readContracts.ts

```diff
@@ -42,17 +46,17 @@ export async function readContracts<Data extends any[] = Result[]>({
         [chainId]: [...(contracts[chainId] || []), contract],
       }
     }, {})
-    const promises = Object.entries(contractsByChainId).map(
-      ([chainId, contracts]) =>
+    const promises = () =>
```

Comment:
We were originally invoking the promises instantly, instead of deferring it to the `Promise.all`.